manifests/server/combine_certs.pp

```diff
@@ -0,0 +1,21 @@
+# combine the public x509 certificate and private key into one file
+#
+define puppet::server::combine_certs(
+  $combined = undef,
+  $cert     = undef,
+  $key      = undef,
+) {
+  $dir = dirname($combined)
+
+  $file_cert = file($cert)
```

Comment:
This is probably not going to work. Here you assume the file is served as a static file from inside this module. Another problem is that this is done server side while the passenger setup often just assumes local certificates exist. Worst case you're going to end up with a puppet server side certificate (SSLProxyMachineCertificateFile) on the puppet agent side. That can be a security issue.

You'll likely need to use https://github.com/puppetlabs/puppetlabs-concat to combine the files.

Reply:
I was testing on Red Hat's Satellite 6.3 which supports puppet 3 and 4. puppetlabs-concat requires > puppet 4.7.0. I was hoping to use something that was compatible with both. I guess that whatever module creates the certificates as part of the install, should probably create the concatenated file that SSLProxyMachineCertificateFile requires?

Reply:
There are versions of puppetlabs-concat that allow older versions but have the same API. We already rely on it heavily. The thing is that Puppet 3 is really deprecated and running with passenger is as well. With Puppet 6 around the corner I'm more leaning to removing the support than fixing any issues left.

Satellite 6.4 will drop Puppet 3 support as well and require you to update to Puppet 4 while you're on Satellite 6.3.

It is true that we don't have similar proxy code for use with puppetserver though so maybe the question is how we can facilitate that if needed.